django_mongodb/compiler.py

```diff
@@ -33,11 +35,21 @@ def execute_sql(
         except EmptyResultSet:
             return iter([]) if result_type == MULTI else None
 
-        return (
-            (self._make_result(row, columns) for row in query.fetch())
-            if result_type == MULTI
-            else self._make_result(next(query.fetch()), columns)
-        )
+        cursor = query.get_cursor()
+        if result_type == SINGLE:
```

Comment:
@WaVEV I pulled in the `if result_type == SINGLE:` block from https://github.com/mongodb-labs/django-mongodb/pull/84. Please check this patch doesn't break anything for you there. This block isn't used until that PR.

Reply:
it seems equivalent.